app/tests/spec/views/complete_account_unlock.js

```diff
@@ -0,0 +1,219 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use strict';
+
+
+define([
+  'chai',
+  'sinon',
+  'lib/promise',
+  'views/complete_account_unlock',
+  'lib/auth-errors',
+  'lib/metrics',
+  'lib/constants',
+  'lib/fxa-client',
+  'lib/url',
+  'models/reliers/relier',
+  'models/auth_brokers/base',
+  'models/user',
+  '../../mocks/router',
+  '../../mocks/window',
+  '../../lib/helpers'
+],
+function (chai, sinon, p, View, AuthErrors, Metrics, Constants, FxaClient,
+      Url, Relier, Broker, User, RouterMock, WindowMock, TestHelpers) {
+  var assert = chai.assert;
+
+  describe('views/complete_account_unlock', function () {
+    var view;
+    var routerMock;
+    var windowMock;
+    var accountUnlockError;
+    var metrics;
+    var fxaClient;
+    var relier;
+    var broker;
+    var user;
+    var account;
+    var validCode = TestHelpers.createRandomHexString(Constants.CODE_LENGTH);
+    var invalidCode = TestHelpers.createRandomHexString(Constants.CODE_LENGTH - 1);
+    var validUid = TestHelpers.createRandomHexString(Constants.UID_LENGTH);
+    var invalidUid = TestHelpers.createRandomHexString(Constants.UID_LENGTH - 1);
+
+    function testShowsExpiredScreen(search) {
+      windowMock.location.search = search || '?code=' + validCode + '&uid=' + validUid;
+      initView();
+      return view.render()
+        .then(function () {
+          assert.ok(view.$('#fxa-account-unlock-link-expired-header').length);
+          testEventLogged('complete-account-unlock.link_expired');
+        });
+    }
+
+    function testShowsDamagedScreen(search) {
+      windowMock.location.search = search || '?code=' + validCode + '&uid=' + validUid;
```

Comment:
I took a long time understand the default value for `search` param here.  It seems to be constructing a valid URL, but you depend on setting an error in the environment to actually trigger the damaged-link error.  AFAICT you only call it with a non-damanged `search` param once, perhaps it would be clearer to remove the default value here and set it explicitly in the test that uses it.